core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -207,6 +207,10 @@ package object config {
     .booleanConf
     .createWithDefault(false)
 
+  private[spark] val APP_CALLER_CONTEXT = ConfigBuilder("spark.log.callerContext")
```

Comment:
Personally I don't like using the spark.hadoop. for this if we know its already being used and these configs will always be added to the hadoop config when not needed.  I guess perhaps we messed up naming that to put things into hadoop configuration.

what about spark.log.hadoop.callerContext?  Although perhaps we should set a policy for this in general.

I would say we should change spark.hadoop being applied to hadoop conf (to like spark.hadoopConf.) but even though I don't see it documented anywhere I think it would be to painful to change  as I know people are using it.